pkg/test/config/toolchainconfig.go

```diff
@@ -107,6 +107,38 @@ func (o AutomaticApprovalOption) MaxNumberOfUsers(overall int, perMember ...PerM
 	return o
 }
 
+type CapacityThresholdsOption struct {
+	*ToolchainConfigOptionImpl
+}
+
+func CapacityThresholds() *CapacityThresholdsOption {
+	c := &CapacityThresholdsOption{
+		ToolchainConfigOptionImpl: &ToolchainConfigOptionImpl{},
+	}
+	return c
+}
+
+func (c CapacityThresholdsOption) ResourceCapacityThreshold(defaultThreshold int, perMember ...PerMemberClusterOptionInt) CapacityThresholdsOption {
```

Comment:
shouldn't it be `(c *CapacityThresholdsOption)`, since the method calls `c.addFunction(..)` which is defined as:
```
func (option *ToolchainConfigOptionImpl) addFunction(funcToAdd ToolchainConfigOptionFunc)
```

Reply:
unit tests are passing, so I think it will work either way.

I see that `ToolchainConfigOptionImpl` has pointer receiver for `Apply` and `addFunction` which makes sense since:
- `addFunction` it's appending functions to `option.toApply` function.
- `Apply` is applying all the functions from `option.toApply`

about `ResourceCapacityThreshold` and `MaxNumberOfSpaces` I'm not sure what will be the reasoning behind it, since they are just wrappers and not modifying the _receiver_ .
could you elaborate more? 
Is because of smaller memory footprint or readability (but we will potentially loose _immutability_ ) ?

Reply:
I'm sure your code is valid, but it's a matter of readability indeed. It seems more logical to me that we manipulate a `*CapacityThresholdsOption` since we're modifying its content when calling its methods.

Reply:
ok thanks for the feedback! 
Could you also explain more about why would be better to have  _no pointer to ToolchainConfigOptionImpl_ in this case. I'm referring to this part of code from your suggestion:
```
// no pointer to ToolchainConfigOptionImpl
type CapacityThresholdsOption struct {
  ToolchainConfigOptionImpl 
}
```

Reply:
Ok got it!

Maybe we can also consider a different naming ( same separate refactor PR maybe):

from `CapacityThresholds.ResourceCapacityThreshold()` to something like `CapacityThresholds.Resources()` or something more tidy. This also means renaming `ResourceCapacityThreshold` to just `Resources`. 
If makes sense for everyone.

Reply:
@xcoulon are you ok with addressing this refactor in a separate PR ?

Reply:
sure, let's change that in a separate PR so this doesn't block this one.